src/Microsoft.AspNetCore.Razor/Compilation/TagHelpers/TagHelperDescriptorComparer.cs

```diff
@@ -61,7 +62,8 @@ public virtual bool Equals(TagHelperDescriptor descriptorX, TagHelperDescriptor
                     descriptorX.AllowedChildren.OrderBy(child => child, StringComparer.OrdinalIgnoreCase),
                     descriptorY.AllowedChildren.OrderBy(child => child, StringComparer.OrdinalIgnoreCase),
                     StringComparer.OrdinalIgnoreCase))) &&
-                descriptorX.TagStructure == descriptorY.TagStructure;
+                descriptorX.TagStructure == descriptorY.TagStructure &&
+                descriptorX.PropertyBag.OrderBy(x => x.Key).SequenceEqual(descriptorY.PropertyBag.OrderBy(y => y.Key));
```

Comment:
- Do Enumerable.SequenceEqual to stay consistent with the above.
- Provide Comparer in both order bys
- `propertyX`, `propertyY`